Cabal/Distribution/Package.hs

```diff
@@ -18,7 +18,7 @@
 
 module Distribution.Package (
         -- * Package ids
-        PackageName(..),
+        PackageName, unPackageName, mkPackageName,
```

Comment:
I sympathise, but I'd like to have that in a separate PR, where we can formalise what "safe" entails (we've uses of `PackageName` values which are deliberately an empty string)